### PR TITLE
feat(payment): PI-1564 Refactor the existing `Moneris` components in …

### PR DESCRIPTION
**Pull request by bc-sebastianszafraniec**
…Checkout JS to use the new checkout payment integration JS API

## What?
Create Moneris package in checkout-js-js and move to it Moneris payment strategy.

## Why?
For better separation between payment integrations and core Checkout JS

## Testing / Proof
Zrzut ekranu 2024-10-31 o 13 50 16

https://github.com/user-attachments/assets/82a681fa-37b6-4fc0-9146-a4db97ec94c7

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/moneris-integration/tsconfig.spec.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "include": [
+    "**/*.test.ts",
+    "**/*.spec.ts",
+    "**/*.test.tsx",
+    "**/*.spec.tsx",
+    "**/*.test.js",
+    "**/*.spec.js",
+    "**/*.test.jsx",
+    "**/*.spec.jsx",
```
Comment (serhii-tkachenko) [inline packages/moneris-integration/tsconfig.spec.json:16]: js and jsx related rows can be removed since we don't use them in this project

**Review by serhii-tkachenko — COMMENTED** [review]
Overall looks good to me. May I ask you to check if moneris-integration package test coverage >= 80%?

File: packages/moneris-integration/tsconfig.spec.json (hunk 1 above)
Comment (bc-sebastianszafraniec) [reply packages/moneris-integration/tsconfig.spec.json]: The coverage for moneris-integration package is meet. Attached screenshot in PR description shows test run executed with `npx nx run moneris-integration:test --coverage`

**Review by serhii-tkachenko — APPROVED** [review]
Looks good to me 👍

File: packages/moneris-integration/tsconfig.lib.json
Comment (bc-peng) [inline packages/moneris-integration/tsconfig.lib.json:1]: EOF

We might not need this file actually.

**Review by bc-peng — APPROVED** [review]

**Pull request merged**